cirq/work/observable_measurement.py

```diff
@@ -0,0 +1,210 @@
+# Copyright 2020 The Cirq developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import dataclasses
+import itertools
+from typing import Iterable, List, Tuple, TYPE_CHECKING, Sequence, Dict
+
+import numpy as np
+import sympy
+
+from cirq import circuits, ops, value
+from cirq.work.observable_settings import (
+    InitObsSetting,
+    _MeasurementSpec,
+)
+
+if TYPE_CHECKING:
+    import cirq
+    from cirq.value.product_state import _NamedOneQubitState
+
+
+def _with_parameterized_layers(
+    circuit: 'cirq.Circuit',
+    qubits: Sequence['cirq.Qid'],
+    no_initialization: bool,
+) -> 'cirq.Circuit':
+    """Return a copy of the input circuit with a parameterized single-qubit
+    rotations.
+
+    These rotations flank the circuit: the initial two layers of X and Y gates
+    are given parameter names "{qubit}-Xi" and "{qubit}-Yi" and are used
+    to set up the initial state. If `no_initialization` is set to True,
+    these two layers of gates are omitted.
+
+    The final two layers of X and Y gates are given parameter names
+    "{qubit}-Xf" and "{qubit}-Yf" and are use to change the frame of the
+    qubit before measurement, effectively measuring in bases other than Z.
+    """
+    x_beg_mom = ops.Moment([ops.X(q) ** sympy.Symbol(f'{q}-Xi') for q in qubits])
+    y_beg_mom = ops.Moment([ops.Y(q) ** sympy.Symbol(f'{q}-Yi') for q in qubits])
+    x_end_mom = ops.Moment([ops.X(q) ** sympy.Symbol(f'{q}-Xf') for q in qubits])
+    y_end_mom = ops.Moment([ops.Y(q) ** sympy.Symbol(f'{q}-Yf') for q in qubits])
+    meas_mom = ops.Moment([ops.measure(*qubits, key='z')])
+    if no_initialization:
+        total_circuit = circuit.copy()
+    else:
+        total_circuit = circuits.Circuit([x_beg_mom, y_beg_mom])
+        total_circuit += circuit.copy()
+    total_circuit.append([x_end_mom, y_end_mom, meas_mom])
+    return total_circuit
+
+
+_PAULI_TO_PARAM_VAL: Dict[Tuple['cirq.Pauli', bool], Tuple[float, float]] = {
```

Comment:
Maybe 
```suggestion
_OBS_TO_PARAM_VAL: Dict[Tuple['cirq.Pauli', bool], Tuple[float, float]] = {
```
? 
It took me a second to figure out what this is. Maybe add a bit of comment here.

Reply:
changed the name, added a docstring, added a short docstring to the next mapping too